actions/publish-github-pages/entrypoint.sh

```diff
@@ -0,0 +1,92 @@
+    cp -r ./hugo/content/ ./customer-docs/
+    cp -r ./hugo/Dockerfile ./customer-docs/
+    cp -r ./hugo/.gitignore ./customer-docs/
+    cp ./hugo/config.yaml ./customer-docs/
+    cp ./hugo/.htmltest.yml ./customer-docs/
+    cp ./hugo/Makefile ./customer-docs/
+    # The following two lines can be removed once this branch is merged into master
+    sed -i 's/yq eval/yq -M eval/' ./customer-docs/Makefile # this can be removed once this branch is merged into master
+    sed -i 's/^export DOCKER_RUN_FLAGS ?= -it --rm$/export DOCKER_RUN_FLAGS ?= --rm/' ./customer-docs/Makefile
+    
+    # copy all customer documentation into the build folder
+    IFS="," read -r -a CONTENT_ARRAY <<< "$CONTENT"
+    for content in "${CONTENT_ARRAY[@]}"; do
+        # clear files needed for storing intermediate variables
+        rm file_origins.txt file_destinations.txt
+        # rename all `README.md` to `_index.md`
+        find ${GITHUB_PAGES_PULL_PATH}${content} -type f -name README.md -print0 | xargs --null -I{} bash -c 'mv {} $(dirname {})/_index.md'
+        # categories with no subfolders, and only a single `_index.md`: `mv foobar/_index.md foobar.md`
+        find ${GITHUB_PAGES_PULL_PATH}${content} -type f -name _index.md -print0 | xargs --null -I{} bash -c 'if [ "$(ls -1q $(dirname {}) | wc -l)" == "1" ]; then echo "$(dirname {})"; mv {} $(dirname $(dirname {}))/$(basename $(dirname {})).md; rm -r $(dirname {}); ls $(dirname $(dirname {})); fi'
+        # install the customer docs (.md pages) to the content folder
+        find ${GITHUB_PAGES_PULL_PATH}${content} -type f -name "*.md" >> file_origins.txt
+        find ${GITHUB_PAGES_PULL_PATH}${content} -type f -name "*.md" -print0 | xargs --null -I{} bash -c 'echo "./customer-docs/content/{}"' | sed -e "s|$GITHUB_PAGES_PULL_PATH||" >> file_destinations.txt
+        readarray -t FILE_ORIGINS < file_origins.txt
+        readarray -t FILE_DESTINATIONS < file_destinations.txt
+        for i in "${!FILE_ORIGINS[@]}"; do
+            echo "Copying ${FILE_ORIGINS[i]} to ${FILE_DESTINATIONS[i]}."
+            mkdir -p ${FILE_DESTINATIONS[i]} && cp ${FILE_ORIGINS[i]} ${FILE_DESTINATIONS[i]}
+        done
+    done
+    
+    # Build Docker image needed to build the Hugo site
+    cd ./customer-docs
+    docker build -t cloudposse/docs .
+
+    # publish the Hugo-generated HTML to $GITHUB_PAGES_PATH
+    make release
+    make real-clean hugo/build
+    cp -r ${HUGO_PUBLISH_DIR} ${GITHUB_PAGES_PUSH_PATH}
+    
+    # commit the newly-generated customer docs website to the customer docs repo
+    git config --global user.email "github-actions-runner@cloudposse.com"
+    git config --global user.name "github-actions-runner"
+    git -C $GITHUB_PAGES_PUSH_PATH add -A
+    git -C $GITHUB_PAGES_PUSH_PATH commit -a --message 'Updating content to $GIT_REF'
+    git -C $GITHUB_PAGES_PUSH_PATH push https://${GITHUB_ACTOR}:${GITHUB_TOKEN}@$(sed "s/https\?:\/\///" <<< ${GITHUB_PAGES_REPO}).git
```

Comment:
Maybe we should do this instead as an action step:

https://github.com/cloudposse/actions/tree/master/github/git-push

Reply:
I'm stoked that this action exists, but it doesn't have any `git -C $DIR` functionality. It only pushes from the runner's main directory. Two possible solutions:
1) rework my script to use the main dir of the github actions runner for the github pages branch of the customer's repo
2) upgrade the `git-push` action to have `git -C $DIR` capabilities.

I happily do the altter; I just want to make sure that's ok.

Reply:
The `cloudposse/actions` `git-push` action ended up getting a serious makeover in order to accommodate our current use case:
https://github.com/cloudposse/actions/pull/107
https://github.com/cloudposse/actions/pull/109